packages/sdk-react/src/core/queries/useCounterpart.ts

```diff
@@ -554,93 +374,84 @@ export const useUpdateCounterpartContact = (counterpartId: string) => {
 ) => {
-  const { monite } = useMoniteContext();
-
-  return useQuery<CounterpartPaginationResponse, Error>({
-    queryKey: [...counterpartQueryKeys.list(), ...args] as const,
-    queryFn: () => monite.api.counterparts.getList(...args),
-  });
+  const { api } = useMoniteContext();
+  return api.counterparts.getCounterparts.useQuery(parameters ?? {});
 };
 
 export const useCreateCounterpart = () => {
   const { i18n } = useLingui();
-  const { monite } = useMoniteContext();
-  const { invalidate } = useCounterpartListCache();
-  const { setEntity } = useCounterpartDetailCache();
-
-  return useMutation<CounterpartResponse, Error, CounterpartCreatePayload>({
-    mutationFn: (payload) => monite.api.counterparts.create(payload),
-
-    onSuccess: (counterpart) => {
-      setEntity(counterpart);
-      invalidate();
-
-      toast.success(
-        t(i18n)`Counterpart “${getCounterpartName(counterpart)}” was created.`
-      );
-    },
+  const { api } = useMoniteContext();
+  const queryClient = useQueryClient();
+
+  return api.counterparts.postCounterparts.useMutation(
+    {},
+    {
+      onSuccess: async (counterpart) => {
+        await api.counterparts.getCounterpartsId.invalidateQueries(queryClient);
+        toast.success(
+          t(i18n)`Counterpart “${getCounterpartName(counterpart)}” was created.`
+        );
+      },
 
-    onError: () => {
-      toast.error(t(i18n)`Failed to create Counterpart.`);
-    },
-  });
+      onError: () => {
+        toast.error(t(i18n)`Failed to create Counterpart.`);
+      },
+    }
+  );
 };
 
 export const useCounterpartById = (id?: string) => {
-  const { monite } = useMoniteContext();
-
-  return useQuery<CounterpartResponse | undefined, Error>({
-    queryKey: counterpartQueryKeys.detail(id),
+  const { api } = useMoniteContext();
 
-    queryFn: () => (id ? monite.api.counterparts.getById(id) : undefined),
-
-    enabled: !!id,
-  });
+  return api.counterparts.getCounterpartsId.useQuery(
+    {
+      path: {
+        counterpart_id: id ?? '',
+      },
+    },
+    {
+      enabled: !!id,
+    }
+  );
 };
 
 export const useUpdateCounterpart = () => {
   const { i18n } = useLingui();
-  const { monite } = useMoniteContext();
-  const { invalidate } = useCounterpartListCache();
-  const { setEntity } = useCounterpartDetailCache();
-
-  return useMutation<CounterpartResponse, Error, CounterpartUpdate>({
-    mutationFn: ({ id, payload }) =>
-      monite.api.counterparts.update(id, payload),
-
-    onSuccess: (counterpart) => {
-      setEntity(counterpart);
-      invalidate();
-
+  const { api } = useMoniteContext();
+  const queryClient = useQueryClient();
+
+  return api.counterparts.patchCounterpartsId.useMutation(undefined, {
+    onSuccess: async (counterpart) => {
+      await api.counterparts.getCounterpartsId.invalidateQueries(
+        {
+          parameters: { path: { counterpart_id: counterpart.id } },
+        },
+        queryClient
+      );
```

Comment:
Here you need to invalidate the list:
```ts
await api.counterparts.getCounterparts.invalidateQueries(queryClient);
```

and instead of `api.counterparts.getCounterpartsId.invalidateQueries` you may directly set the cache:

```ts
api.counterparts.getCounterpartsId.setQueryData(
  {
    path: { counterpart_id: counterpart.id },
  },
  (prevCounterpart) => ({
    ...prevCounterpart,
    ...counterpart,
  }),
  queryClient
);
```